scipy/sparse/tests/test_base.py

```diff
@@ -1146,9 +1146,16 @@ def test_astype(self):
         S = self.spmatrix(D)
 
         for x in supported_dtypes:
-            assert_equal(S.astype(x).dtype, D.astype(x).dtype)  # correct type
-            assert_equal(S.astype(x).toarray(), D.astype(x))        # correct values
-            assert_equal(S.astype(x).format, S.format)           # format preserved
+            # Check correct casted
+            D_casted = D.astype(x)
+            for copy in (True, False):
+                S_casted = S.astype(x, copy=copy)
+                assert_equal(S_casted.dtype, D_casted.dtype)  # correct type
+                assert_equal(S_casted.toarray(), D_casted)    # correct values
+                assert_equal(S_casted.format, S.format)       # format preserved
+            # Check correct copied
+            assert S_casted.astype(x, copy=False) is S_casted
+            assert S_casted.astype(x, copy=True) is not S_casted
```

Comment:
You are right. I added a check if this is deep copy. But it was a little bit more difficult.